src/main/java/SentinelRefactoring.java

```diff
@@ -0,0 +1,369 @@
+			return true;
+		}
+
+		// TODO: Confirm this works with non-ints
+		// The variable previously had the same value as the potential new value
+		Object oldValue = valueExpr.resolveConstantExpressionValue();
+		Object newValue = new_value.resolveConstantExpressionValue();
+		if (oldValue == newValue) {
+			LOGGER.debug("New value of Sentinel \"" + sentinelName + "\" matches old value. The Sentinel is invalid.");
+			LOGGER.debug("Old Value: " + oldValue);
+			LOGGER.debug("New Value: " + newValue);
+			return false;
+		}
+		return true;
+
+	}
+
+	private void updateSentinelValues(ASTNode node) {
+		// TODO: Confirm works with the classes VariableDeclaration,
+		// SingleVariableDeclaration
+		if (node instanceof VariableDeclaration declaration) {
+			updateSentinelValues(declaration);
+		} else if (node instanceof Assignment assign) {
+			updateSentinelValues(assign);
+		} else if (node instanceof MethodInvocation || node instanceof SuperMethodInvocation) {
+			LOGGER.debug("Clearing all sentinel values due to method invocation...");
+			Enumeration<String> keys = sentinel_values.keys();
+			while (keys.hasMoreElements()) {
+				String key = keys.nextElement();
+				sentinel_values.put(key, new Sentinel_Value(false, null));
+			}
+		}
+	}
+
+	private void updateSentinelValues(VariableDeclaration declaration) {
+		sentinel_values.put(declaration.getName().toString(), new Sentinel_Value(false, declaration.getInitializer()));
+	}
+
+	private void updateSentinelValues(Assignment statement) {
+		if (!(statement.getLeftHandSide() instanceof SimpleName varName)) {
+			return;
+		}
+
+		if (sentinel_values.get(varName.toString()) != null) {
+			sentinel_values.put(varName.toString(), new Sentinel_Value(false, statement.getRightHandSide()));
+		}
+	}
+
+	@Override
+	public boolean isApplicable(ASTNode node) {
+		if (skipSentinelDeclaration) {
+			if (node instanceof VariableDeclarationStatement || node instanceof Assignment) {
+				skipSentinelDeclaration = false;
+			}
+			return false;
+		}
+		if (node instanceof IfStatement ifStmt) {
+			if (isApplicable(ifStmt)) {
+				updateSentinelValues(node);
+				return true;
+			}
+		}
+		updateSentinelValues(node);
+		if (node instanceof Assignment assign) {
+			checkReassignment(assign);
+		} else if (node instanceof VariableDeclarationStatement declaration) {
+			checkShadowing(declaration);
+		}
+		return false;
+
+	}
+
+	public boolean isApplicable(IfStatement ifStmt) {
+		List<Expression> exprs = Refactoring.getSubExpressions(ifStmt.getExpression());
+		for (Expression expression : exprs) {
+			if (!(expression instanceof InfixExpression infix)) {
+				continue;
+			}
+
+			// Checks if node represents a check on a known sentinel value
+			if (isApplicable(infix)) {
+				return true;
+			}
+
+			// Parse through infix statement to see if there are any sentinels
+			parseSentinels(infix, ifStmt);
+		}
+		return false;
+	}
+
+	// Todo: Confirm this still works
```

Comment:
Is this done? Please go through and check all of these before asking me to review.